From: John Johansen <john@jjmx.net>
Date: Tue, 22 Feb 2022 19:53:48 +0000
Subject: Merge GTK abstraction: add support for gtk4.

The recently added gtk abstraction (!825) lack the support for the new gtk4.

MR: https://gitlab.com/apparmor/apparmor/-/merge_requests/857
Approved-by: Jon Tourville <jon.tourville@canonical.com>
Merged-by: John Johansen <john@jjmx.net>

(cherry picked from commit 6f18326232a471557ed7492e91a0cec94c0fb0cb)

8074ff4f GTK abstraction: add support for gtk4.
---
 profiles/apparmor.d/abstractions/gtk | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/profiles/apparmor.d/abstractions/gtk b/profiles/apparmor.d/abstractions/gtk
index 9f12764..26c388b 100644
--- a/profiles/apparmor.d/abstractions/gtk
+++ b/profiles/apparmor.d/abstractions/gtk
@@ -16,14 +16,14 @@
   /usr/share/gtk-2.0/ r,
   /usr/share/gtk-2.0/gtkrc r,
 
-  /usr/share/gtk-3.0/ r,
-  /usr/share/gtk-3.0/settings.ini r,
+  /usr/share/gtk-{3,4}.0/ r,
+  /usr/share/gtk-{3,4}.0/settings.ini r,
 
   /etc/gtk-2.0/ r,
   /etc/gtk-2.0/gtkrc r,
 
-  /etc/gtk-3.0/ r,
-  /etc/gtk-3.0/*.conf r,
+  /etc/gtk-{3,4}.0/ r,
+  /etc/gtk-{3,4}.0/*.conf r,
 
   /etc/gtk/gtkrc r,
 
@@ -36,10 +36,10 @@
   owner @{HOME}/.gtk-bookmarks r,
   owner @{HOME}/.config/gtkrc r,
   owner @{HOME}/.config/gtkrc-2.0 r,
-  owner @{HOME}/.config/gtk-3.0/ rw,
-  owner @{HOME}/.config/gtk-3.0/settings.ini r,
-  owner @{HOME}/.config/gtk-3.0/bookmarks r,
-  owner @{HOME}/.config/gtk-3.0/gtk.css r,
+  owner @{HOME}/.config/gtk-{3,4}.0/ rw,
+  owner @{HOME}/.config/gtk-{3,4}.0/settings.ini r,
+  owner @{HOME}/.config/gtk-{3,4}.0/bookmarks r,
+  owner @{HOME}/.config/gtk-{3,4}.0/gtk.css r,
 
   # for gtk file dialog
   owner @{HOME}/.config/gtk-2.0/ rw,
